.swm/state-management-with-reflux.jni2e.sw.md

```diff
@@ -0,0 +1,193 @@
+---
+id: jni2e
+name: State Management with Reflux
+file_version: 1.0.2
+app_version: 0.8.5-0
+file_blobs:
+  static/app/actions/pageFiltersActions.tsx: 8d6f2485c157b80586df789ae5c655bf51299ec7
+  static/app/stores/pageFiltersStore.tsx: c04a27b5e33fd62b865b58b4bddac1be1b899f3e
+  static/app/actionCreators/pageFilters.tsx: 2b27c6d727455f7dde4fa3ea8210a4b6162b661f
+  static/app/types/core.tsx: c27f92c56ab334744f9a5a39990b7b86e6c1aa27
+---
+
+We currently use [Reflux](https://github.com/reflux/refluxjs) for managing global state. Reflux is a simple library for unidirectional dataflow architecture inspired by ReactJS [Flux](http://facebook.github.io/react/blog/2014/05/06/flux.html).
+
+# Overview
+
+The main function of Reflux is to introduce a more functional programming style architecture by adopting a single data flow pattern.
+
+```
++---------+       +--------+       +-----------------+
+¦ Actions ¦------>¦ Stores ¦------>¦ View Components ¦
++---------+       +--------+       +-----------------+
+     ^                                      ¦
+     +--------------------------------------+
+```
+
+The pattern is composed of actions and data stores, where actions initiate new data to pass through data stores before coming back to the view components again. If a view component has an event that needs to make a change in the application's data stores, they need to do so by signaling to the stores through the actions available.
+
+We will use `PageFilters`[<sup id="1HGoLB">↓</sup>](#f-1HGoLB) for our example.
+
+For usage, you need to create actions which can be called from React components. Those actions are listened to by stores which hold and update data. In turn those stores are hooked up to React components and set state within them as it is updated within the store.
+
+So there are three parts:
+
+# 1\. Creating Actions
+
+An action is a [function object](http://en.wikipedia.org/wiki/Function_object) that can be invoked like any other function.
+
+Actions are registered under `📄 static/app/actions` , e.g., `PageFiltersActions`[<sup id="Z2wU2DS">↓</sup>](#f-Z2wU2DS) are registered in `📄 static/app/actions/pageFiltersActions.tsx`
+
+<br/>
+
+To create actions we use `createActions`[<sup id="1Hy98y">↓</sup>](#f-1Hy98y) , like so:
```

Comment:
Unclear to me if this creates all the actions? only a certain group of actions? are there other calls to `createActions?`

Reply:
I actually think this sentence is straightforward. How would you rewrite it?
(It's for all actions, and there are many calls...)